x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java

```diff
@@ -255,6 +259,29 @@ public void validate(Collection<Phase> phases) {
                 MigrateAction.NAME + " action and an " + AllocateAction.NAME + " action with allocation rules. specify only a single " +
                 "data migration in each phase");
         }
+
+        validateActionsFollowingSearchableSnapshot(phases);
+    }
+
+    static void validateActionsFollowingSearchableSnapshot(Collection<Phase> phases) {
+        boolean hotPhaseContainsSearchableSnapshot = phases.stream()
+            .filter(phase -> HOT_PHASE.equals(phase.getName()))
+            .anyMatch(phase -> phase.getActions().containsKey(SearchableSnapshotAction.NAME));
+        if (hotPhaseContainsSearchableSnapshot) {
+            String phasesDefiningIllegalActions = phases.stream()
+                // we're looking for prohibited actions in phases other than hot
+                .filter(phase -> HOT_PHASE.equals(phase.getName()) == false)
+                // filter the phases that define illegal actions
+                .filter(phase ->
+                    Collections.disjoint(ACTIONS_CANNOT_FOLLOW_SEARCHABLE_SNAPSHOT, phase.getActions().keySet()) == false)
+                .map(Phase::getName)
+                .collect(Collectors.joining(","));
+            if (Strings.hasText(phasesDefiningIllegalActions)) {
+                throw new IllegalArgumentException("phases [" + phasesDefiningIllegalActions + "] define one or more of [" +
+                    String.join(",", ACTIONS_CANNOT_FOLLOW_SEARCHABLE_SNAPSHOT) + "] actions which are not allowed after the " +
+                    "managed index was mounted as searchable snapshot");
```

Comment:
Collections already `toString` with commands, so this can be:

```suggestion
                throw new IllegalArgumentException("phases [" + phasesDefiningIllegalActions + "] define one or more of " +
                    ACTIONS_CANNOT_FOLLOW_SEARCHABLE_SNAPSHOT + " actions which are not allowed after a " +
                    "managed index is mounted as a searchable snapshot");
```